src/config.c

```diff
@@ -94,6 +94,13 @@ configEnum aof_fsync_enum[] = {
     {NULL, 0}
 };
 
+configEnum shutdown_savemode_sig_enum[] = {
+    {"auto", 0},
+    {"save", SHUTDOWN_SAVE},
+    {"nosave", SHUTDOWN_NOSAVE},
```

Comment:
But then we can’t combine save / nosave with the other flags, right?
That was the idea when reusing the parser of shutdown command.

The NOW can be achieved by setting shutdown-timeout 0 as well, so I’m not sure we need it.

Reply:
ohh, good point.
maybe we need some config infrastructure for bit flags in addition to enum (probably not now).
or maybe it should follow the footsteps of `maxmemory-policy` (which also combines several flags into one enum).
regarding `shutdown-timeout`, i think the power of the new configs you add now, is that it enables certain signals to have behavior that's different than the default.

Reply:
i gave it some more thought.
i think adding a parser that's specific to the shutdown arguments is wrong.

maybe we could somehow easily leverage the `createEnumConfig` mechanism to allow some enum configs to be bit flags?
i.e. you define the same `configEnum` table, and pass some flag to createEnumConfig which tells the infra to handle things differently.

wanna look into that and check how complicated it is?

I'm not sure we can proceed with this in any other way.
i.e. the alternatives are:
1. what you have now which only controls the `save` feature.
2. add the other options (force, now), but don't allow combining them.
3. use createSpecialConfig, take a string and do all the magic there.
4. start with either of the above, and plan to introduce the bit flags mechanism later in a backwards compatible way.

Reply:
Extending later sounds like a good idea. We need to discuss how multiple flags would be supported and things like that (i.e with spaces or pipes).
In any case then we will need to change the name of this config.
It won’t be only related to saving anymore.
shutdown-mode-sigint, shutdown-options-sigint…
Ideas?

Reply:
maybe we don't even need to mention the term `options`, `mode`, `flags` in the name.
how about just `shutdown-on-sigint` (being set to: `default`, `save`, `nosave`, `force`, `now`)

i don't think it'll be hard to decide on a separator, it could either be space, or `|`.
so we'll go with option [2] i listed above, i.e. add `force` and `now`, but don't allow combining them yet.
and then when we'll do that we'll be backwards compatible with what we introduce today.